packages/discord-bot/src/utils/stateEmbed.ts

```diff
@@ -0,0 +1,39 @@
+import { MessageEmbed, StringMappedInteractionTypes } from 'discord.js';
+import { UserState } from '../interfaces/UserState';
+
+const formatDate = (date: Date): string =>
+  `${date.getDate()}-${date.getMonth()}-${date.getFullYear()}`;
+
+export const getStateEmbed = (state: UserState): MessageEmbed => {
+  const embed = new MessageEmbed()
+    .setTitle(state.username)
+    .setDescription(
+      state.hasPraiseGiverRole && state.activated
+        ? 'Your account is activated and has praise powers.'
+        : state.hasPraiseGiverRole
+        ? "You have praise powers, however your account isn't activated(use `/activate` command)."
+        : "Your account is activated, however you don't have praise powers."
+    )
+    .setThumbnail(
+      `https://cdn.discordapp.com/avatars/${state.id}/${state.avatar || 'oops'}`
+    );
+  if (state.activated) {
+    embed.addField('User Roles', state.praiseRoles?.join(' | ') || 'oops');
```

Comment:
Why is the `|| 'oops'` necessary? Should it be "No roles" or something more specific? Or just blank?

Reply:
We could have "No Roles", though the API assigns everyone the "USER" role on the activation Maybe we can change this up a bit and just not create this field if there are no roles, rather than filling it in with text